x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/InferenceToXContentCompressorTests.java

```diff
@@ -35,18 +31,11 @@ public void testInflateAndDeflate() throws IOException {
     public void testInflateTooLargeStream() throws IOException {
         TrainedModelDefinition definition = TrainedModelDefinitionTests.createRandomBuilder().build();
         String firstDeflate = InferenceToXContentCompressor.deflate(definition);
-        BytesReference inflatedBytes = InferenceToXContentCompressor.inflate(firstDeflate, 10L);
-        assertThat(inflatedBytes.length(), equalTo(10));
-        try(XContentParser parser = XContentHelper.createParser(xContentRegistry(),
-            LoggingDeprecationHandler.INSTANCE,
-            inflatedBytes,
-            XContentType.JSON)) {
-            expectThrows(IOException.class, () -> TrainedModelConfig.fromXContent(parser, true));
-        }
+        expectThrows(IOException.class, () -> Streams.readFully(InferenceToXContentCompressor.inflate(firstDeflate, 10L)));
```

Comment:
Should we also verify that the IOException has the message containing "input stream exceeded maximum bytes"?